.changeset/brown-rocks-ring.md

```diff
@@ -0,0 +1,15 @@
+---
+'astro': major
+---
+
+add `checked` to htmlBooleanAttributes.
+
+Previously, `checked={true}` returned `checked="true"`, which is considered [invalid](https://html.spec.whatwg.org/#boolean-attributes).
+Now, this should return `checked` only. 
+Note: Falsy values would omit the attribute, which is correct behavior.
+
+You'll only need to edit your existing code if you are checking the `checked` attribute's value to be the string `'true'` like so:
+```js
+- el.getAttribute('checked') === 'true'
++ el.hasAttribute('checked')
+```
```

Comment:
I don't think this change will result in a major release as we're only fixing a bug here. You can follow the format like in this similar issue instead which I think should be enough: https://github.com/withastro/astro/pull/12027/files#diff-81fba6c45d17cfb15c45081cbaf691ab9dd03f27ad01ba2db5de791366be6df0

Reply:
Oh I was wondering about that because running `changeset` only let me select a major version. Can I just edit the changeset file to mark this as a patch?

Reply:
Yeah you can edit it directly. I think the CLI has confused others often, but on the step where it requests for major, you can press "Enter" directly to skip to the next prompt that asks for minor, and then patch.